lib/msf/core/option_container.rb

```diff
@@ -208,7 +208,7 @@ def validate(datastore)
       end
 
       # Validate all permutations of rhost combinations
-      if include?('RHOSTS')
+      if include?('RHOSTS') && !(datastore['RHOSTS'].blank? && !get('RHOSTS').required)
```

Comment:
This tweaks the logic to skip the validation when the option is both blank and optional.